### PR TITLE
python tests: add VFIO_USER_DEVICE_SET_IRQS

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: test/py/test_device_set_irqs.py [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    hdr = vfio_user_header(VFIO_USER_DEVICE_SET_IRQS, size=len(payload))
+    sock.sendmsg([hdr + payload], [(socket.SOL_SOCKET, socket.SCM_RIGHTS,
+                 struct.pack("I", fd))])
+    vfu_run_ctx(ctx)
+    get_reply(sock, expect=errno.EINVAL)
+
+    os.close(fd)
+
+def test_device_set_irqs_disable():
+    # struct vfio_irq_set
```
Comment (tmakatos) [inline test/py/test_device_set_irqs.py:225]: Can we have a very short explanation what is achieved with each command? I recall the IRQ API is not trivial.

**Review by tmakatos — APPROVED** [review]
LGTM.

There is a lot of scope for code deduplication, e.g. the 
```  payload = struct.pack(...)
    hdr = vfio_user_header(..., size=len(payload))
    sock.send(hdr + payload)
    vfu_run_ctx(ctx)
    get_reply(...)
```
can be put into a function and most test cases will become one-liners.

File: test/py/test_device_set_irqs.py [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+    ret = vfu_realize_ctx(ctx)
+    assert ret == 0
+
+    sock = connect_client(ctx)
+
+def test_device_set_irqs_short_write():
+    payload = struct.pack("II", 0, 0)
+
+    hdr = vfio_user_header(VFIO_USER_DEVICE_SET_IRQS, size=len(payload))
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/test_device_set_irqs.py:63]: that's a good idea; OK if I do it as a follow-up at the end of the series?

File: test/py/libvfio_user.py [hunk 3]
```diff
@@ -160,8 +167,7 @@
 build_type = os.getenv("BUILD_TYPE", default="dbg")
 libname = "%s/build/%s/lib/libvfio-user.so" % (topdir, build_type)
 lib = c.CDLL(libname, use_errno=True)
-
-msg_id = 1
+libc = c.CDLL("libc.so.6", use_errno=True)
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/libvfio_user.py:170]: unfortunately CDLL doesn't know how to parse the linker script that is libc.so
On the plus side, they're never, ever, going to rev libc again :)

File: test/py/test_device_set_irqs.py [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+                          VFIO_IRQ_SET_DATA_NONE, VFU_DEV_REQ_IRQ, 0, 0)
+
+    hdr = vfio_user_header(VFIO_USER_DEVICE_SET_IRQS, size=len(payload))
+    sock.send(hdr + payload)
+    vfu_run_ctx(ctx)
+    get_reply(sock, expect=errno.EINVAL)
+
+def test_device_set_irqs_bad_index():
+    # struct vfio_irq_set
+    payload = struct.pack("IIIII", 20, VFIO_IRQ_SET_ACTION_TRIGGER |
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/test_device_set_irqs.py:80]: This is much clearer in the later cleanups

File: test/py/test_device_set_irqs.py (hunk 1 above)
Comment (jlevon) [reply test/py/test_device_set_irqs.py:225]: I'm not sure I want to repeat what the spec says? That should hopefully be sufficient?

File: test/py/test_device_set_irqs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tmakatos) [reply test/py/test_device_set_irqs.py:63]: sure

File: test/py/test_device_set_irqs.py (hunk 1 above)
Comment (tmakatos) [reply test/py/test_device_set_irqs.py:225]: I was thinking something super short like "start=0 count=0 means disable all IRQs" but if you think it's redundant we can skip it

**Pull request merged**